spec/setup/helpers/perf_helper_spec.rb

```diff
@@ -49,5 +72,429 @@ def options
 
   end
 
+  context '.setup_r10k' do
+
+    let!(:masters) {[{'platform' => Beaker::Platform.new('centos-6.5-x86_64')}]}
+
+    it 'adds the private key, configures ssh, and installs git on each master' do
+
+      allow(subject).to receive(:masters).and_return(masters)
+      expect(subject).to receive(:select_hosts).with({:roles => ['master', 'compile_master']}).and_return(masters)
+
+      expect(subject).to receive(:create_remote_file).with(masters[0], "/root/.ssh/id_rsa", /PRIVATE/ )
+      expect(subject).to receive(:create_remote_file).with(masters[0], "/root/.ssh/config", /StrictHostKeyChecking/ )
+
+      expect(subject).to receive(:on).with(masters[0], "chmod 600 /root/.ssh/id_rsa /root/.ssh/config" )
+      expect(subject).to receive(:install_package).with(masters[0], 'git')
+      subject.setup_r10k
+
+    end
+
+  end
+
+  # TODO: refactor to remove duplication
+  describe '#perf_install_pe' do
+
+    context 'when pe is not a pre-aio version' do
+
+      test_is_pre_aio_version = false
+
+      context 'when we are using meep' do
+
+        # TODO: how should @options / @options[:answers] be handled?
+        before {
+          test_options = { :answers => {} }
+          #subject.instance_variable_set(:@options, { })
+          subject.instance_variable_set(:@options, test_options)
```

Comment:
i'm fine using this in tests

Reply:
I expected to see `@options[:answers]` updated with `r10k_remote` and `r10k_private_key` but when I check with instance_variable_get it doesn't seem to have been updated. I added some puts to perf_install_pe after it sets those values and it appears to be updated when output to the console. I'm probably doing it wrong.

Reply:
Ah, sorry you're right. I think what you did on line 124 should work, but you need to move it to after the call to subject.perf_install_pe.
You can put the other assertions before that because they have to do with what methods will be called and what params, but if you want to check that values of instance variables changed after executing the method, you need to make sure you assert on those after the method is actually called.

Reply:
Ah, that makes sense, thanks!